nntrainer/graph/network_graph.cpp

```diff
@@ -212,98 +214,90 @@ int NetworkGraph::realizeMultiOutputType(
   return status;
 }
 
-/**
- * @fixme: the implementation assumes loss layer should always be at the last
- * layer and the there is only one loss, this assumption is not true
- */
 int NetworkGraph::addLossLayer(const LossType loss_type) {
   int status = ML_ERROR_NONE;
-  auto const &last_node = LNODE(graph.getSortedNode(graph.size() - 1));
-  auto last_layer_node = getSortedLayerNode(graph.size() - 1);
 
-  if (last_layer_node->getObject()->requireLabel()) {
-    return status;
-  }
+  for (unsigned int i = 0; i < graph.getNumOutputNodes(); ++i) {
+    auto output_layer_node = LNODE(graph.getOutputNode(i));
 
-  if (loss_type == LossType::LOSS_NONE) {
-    return status;
-  }
+    if (output_layer_node->getObject()->requireLabel()) {
+      continue;
```

Comment:
also please add a todo here to realize the cross_entropy loss to softmax/sigmoid cross entropy loss.
This is already implemented in layer_v2, and will be added here upon merge.

Reply:
I think from line 235 to 249 the realization of cross_entropy loss to softmax/sigmoid cross entropy loss exist. Am I misunderstand what you say?

Reply:
If the output_layer_node is cross_entropy layer type, this loop will continue here. However, it also needs realization (not add new loss layer but udpate current loss layer type).

Reply:
Oh now I understand. Thanks